src/Language/PureScript/Docs/Convert/Single.hs

```diff
@@ -100,66 +104,126 @@ getDeclarationTitle (P.ValueFixityDeclaration _ _ _ op) = Just (P.showOp op)
 getDeclarationTitle _ = Nothing
 
 -- | Create a basic Declaration value.
-mkDeclaration :: P.SourceAnn -> Text -> DeclarationInfo -> Declaration
-mkDeclaration (ss, com) title info =
+mkDeclaration :: P.SourceAnn -> Text -> Maybe KindInfo -> DeclarationInfo -> Declaration
+mkDeclaration (ss, com) title kindInfo info =
   Declaration { declTitle      = title
               , declComments   = convertComments com
               , declSourceSpan = Just ss -- TODO: make this non-optional when we next break the format
               , declChildren   = []
               , declInfo       = info
+              , declKind       = kindInfo
               }
 
 basicDeclaration :: P.SourceAnn -> Text -> DeclarationInfo -> Maybe IntermediateDeclaration
-basicDeclaration sa title = Just . Right . mkDeclaration sa title
-
-convertDeclaration :: P.Declaration -> Text -> Maybe IntermediateDeclaration
-convertDeclaration (P.ValueDecl sa _ _ _ [P.MkUnguarded (P.TypedValue _ _ ty)]) title =
-  basicDeclaration sa title (ValueDeclaration (ty $> ()))
-convertDeclaration (P.ValueDecl sa _ _ _ _) title =
-  -- If no explicit type declaration was provided, insert a wildcard, so that
-  -- the actual type will be added during type checking.
-  basicDeclaration sa title (ValueDeclaration (P.TypeWildcard () Nothing))
-convertDeclaration (P.ExternDeclaration sa _ ty) title =
-  basicDeclaration sa title (ValueDeclaration (ty $> ()))
-convertDeclaration (P.DataDeclaration sa dtype _ args ctors) title =
-  Just (Right (mkDeclaration sa title info) { declChildren = children })
+basicDeclaration sa title = Just . Right . mkDeclaration sa title Nothing
+
+reassociateKindSignatures
+  :: (Maybe P.Declaration, [Either (P.Declaration, P.Declaration) P.Declaration])
```

Comment:
Did you consider reusing the existing `DeclarationAugment` mechanism here? I think it could be worth investigating whether we could an a new constructor to `DeclarationAugment` for kind signatures, and add the kind info inside `augmentWith`.

Reply:
Not yet. I was just trying to get the code to build/work first and then get an initial review. I'll look into this.

Reply:
Well that was a whole lot easier...